web/js/map-OpenLayers.js

```diff
@@ -115,6 +115,32 @@ var fixmystreet = fixmystreet || {};
         fixmystreet.markers.redraw();
       },
 
+      show_shortlist_control: function() {
+        var $shortlistButton = $('#fms_shortlist_all');
+        if (fixmystreet.page == "reports") {
+          if (fixmystreet.map.getZoom() >= 14) {
+            $shortlistButton.removeClass('hidden');
+            $shortlistButton.on('click', function() {
+              var features = [];
+              var csrf = $('meta[name="csrf-token"]').attr('content');
+
+              for (var i = 0; i < fixmystreet.markers.features.length; i++) {
+                var feature = fixmystreet.markers.features[i];
+                if (feature.onScreen()) {
+                  features.push(feature.data.id);
+                }
+              }
+
+              $.post( "/my/planned/change_multiple", { ids: features, token: csrf }, function() {
+                $shortlistButton.addClass('hidden');
+              });
```

Comment:
What happens if this POST fails? Some kind of user feedback should be shown, whatever the outcome.

It'd be nice if the sidebar could be updated to show the now-shortlisted reports with the filled-in green star icon.